source/faq.txt

```diff
@@ -148,6 +146,33 @@ When ``MongoClient.close()`` is called by any thread, the driver
 closes all idle sockets and closes all sockets that are in
 use as they are returned to the pool.
 
+
+How do I prevent the "IllegalArgumentException: Invalid BSON field name" error?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Your application may throw this exception when it encounters any of the
+following issues:
+
+- Characters in field names that your MongoDB deployment cannot process
+- Syntax issues in a command
+
+In MongoDB v4.4 and earlier, names that contain characters such as
```

Comment:
I thought this was changed way back in 3.6

Reply:
I haven't checked, but maybe I misread "improved support" as "started support" in the [Server Manual docs](https://www.mongodb.com/docs/manual/core/dot-dollar-considerations/#overview):


> MongoDB 5.0 adds improved support for field names that are dollar ($) prefixed or that contain periods (.). The validation rules for storing data have been updated to make it easier to work with data sources that use these characters.